pandas/tseries/tdi.py

```diff
@@ -754,7 +758,7 @@ def _maybe_cast_slice_bound(self, label, side, kind):
     def _get_string_slice(self, key, use_lhs=True, use_rhs=True):
         freq = getattr(self, 'freqstr',
                        getattr(self, 'inferred_freq', None))
-        if is_integer(key) or is_float(key):
+        if is_integer(key) or is_float(key) or key is tslib.NaT:
```

Comment:
Thanks for your reviews.

> use isnull and don't change internal cython code that should is not relevant

I updated to use `isnull` and do not change on cython codes
https://github.com/pydata/pandas/pull/13687/files#diff-c03d3accabc2e2f441d87b961f4425d7R701

> further we don't use lib.is_integer ever directly 

I use `lib.is_integer` only ipython, this PR does not use `lib.is_integer`, sorry to confuse you.

> iNaT IS by definition an integer

Sure. But this PR converts a key to `NaT` (not `iNaT`) which may not be integer, so we need to check key is `tslib.NaT` or not.